host-go/engine/module/runtime.go

```diff
@@ -9,5 +9,5 @@ type Runtime interface {
 	// NewModule instantiates a new module from the given WAT code.
 	//
 	// This is a fairly expensive operation.
-	NewModule([]byte) (Module, error)
+	NewModule([]byte, bool) (Module, error)
```

Comment:
todo: I dislike that `isWasi` is a mandatory parameter, it complicates the simple-use case.

I think we should either auto-detect whether it is wasi or not, or make the param optional (`...bool`, or similar)

Reply:
If we want to auto-detect we might want another top/runtime-level config-flag/param that disables wasi completely, as otherwise individual modules would be free to escape the wasm-sandbox with out much visibility.

Reply:
I agree it should be optional, in fact it should use a more generic "functional options" like parameter so that we can add more Module options in the future, instead of it being explicitly a optional bool param.

Additionally, I also agree that we should be able to manually configure wasi support at the engine/runtime level, which if disabled, would throw an error if a module is instanciated with the `WithWASI(true)` (example syntax) included as an option.